File: lesson-08/PythonProducer/producer.py
from kafka import KafkaProducer


def produce(topic, servers):
    producer = KafkaProducer(bootstrap_servers=servers, value_serializer=str.encode)

    for i in range(1000):
        producer.send(topic, key=i.to_bytes(length=2, byteorder='big'), value='Message '+str(i))

    producer.flush()


if __name__ == "__main__":
    produce('test', 'localhost:9092')

File: lesson-02/BaseExample/Clojure/producer/src/ru/vzaigrin/examples/kafka/base/clj/producer.clj
(ns ru.vzaigrin.examples.kafka.base.clj.producer
  (:gen-class)
  (:import 
   (java.util Properties)
   (org.apache.kafka.clients.producer KafkaProducer ProducerConfig ProducerRecord)))

(defn -main
  "Base Kafka Producer"
  []
  (let [topic "test"
        props (doto (Properties.)
                (.put ProducerConfig/BOOTSTRAP_SERVERS_CONFIG "localhost:9092")
                (.put ProducerConfig/KEY_SERIALIZER_CLASS_CONFIG "org.apache.kafka.common.serialization.InetgerSerializer")
                (.put ProducerConfig/VALUE_SERIALIZER_CLASS_CONFIG "org.apache.kafka.common.serialization.StringSerializer"))]
    (with-open [producer (KafkaProducer. props)]
      (doseq [i (range 1 1001)] (.send producer (ProducerRecord.  topic i (str "Message " i))))
      (.flush producer))))
